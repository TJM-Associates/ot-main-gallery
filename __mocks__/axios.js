const get = (url) => {
  const restaurantName = url.split('/')[3];
  if (restaurantName === 'willThrowError') {
    return Promise.reject('Unknown error');
  }

  if (restaurantName === 'noImagesHere') {
    return Promise.resolve({
      data: [],
    });
  }

  return Promise.resolve({
    data: [
      {},
      {},
      {},
      {},
      {},
      {},
      {},
      {},
      {},
      {},
      {},
    ],
  });
};

exports.get = get;
