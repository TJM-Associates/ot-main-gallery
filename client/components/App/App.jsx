import React from 'react';
import Gallery from '../Gallery/Gallery.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      images: [],
    };
  }

  componentDidMount() {
    fetch('/benu', {
      headers: {
        'Content-Type': 'application/json',
      },
    })
    .then((response) => response.json())
    .then((images) => {
      this.setState({ images });
    });
  }

  render() {
    return (
      <Gallery images={this.state.images}/> 
    );
  }
}

export default App;
