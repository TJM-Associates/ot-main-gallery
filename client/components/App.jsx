import React from 'react';
import axios from 'axios';
import Gallery from './Gallery.jsx';

const Header = styled.h2`
  padding-bottom: 20px;
  font-family: BrandonText-Black;
`;

Header.displayName = 'Header';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      images: [],
    };
    this.getImages = this.getImages.bind(this);
  }

  componentDidMount() {
    this.getImages();
  }

  getImages() {
    axios.get(`/api${window.location.pathname}images`, {
      headers: {
        'Content-Type': 'application/json',
      },
    })
      .then((response) => {
        this.setState({ images: response.data });
      })
      .catch((err) => {
        console.log(err);
        this.setState({ images: [] });
      });
  }

  render() {
    const { images } = this.state;

    if (images.length === 0) {
      return null;
    }

    return (
      <div>
        <Header>
          {images.length}
          {' '}
Photos
        </Header>
        <Gallery images={images} />
      </div>
    );
  }
}

window.Gallery = App;
export default App;
