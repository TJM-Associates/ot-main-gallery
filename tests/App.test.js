/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import axios from 'axios';
import App from '../client/components/App.jsx';

describe('App', () => {
  it('should render properly, with component did mount calling getImages', () => {
    axios.get = () => (
      Promise.resolve('success')
    );

    // eslint-disable-next-line react/jsx-filename-extension
    let wrapper = shallow(<App />);
    axios.get = () => (
      Promise.reject('failure')
    );
    wrapper = shallow(<App />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
