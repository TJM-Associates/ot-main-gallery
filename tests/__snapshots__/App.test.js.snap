// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly, with component did mount calling getImages 1`] = `
<Gallery
  images={Array []}
/>
`;
